dot/parachain/network-bridge/receiver.go

```diff
@@ -240,46 +248,127 @@ func (nbr *NetworkBridgeReceiver) handleCollationMessage(
+		if !ok {
+			return propagate, fmt.Errorf(
+				"failed to cast into collator protocol message, expected: *CollationProtocol, got: %T",
+				value)
+		}
+		nbr.SubsystemsToOverseer <- events.PeerMessage[collatorprotocolmessages.CollationProtocol]{
+			PeerID:  sender,
+			Message: *collatorProtocol,
+		}
+	case 2:
+		viewUpdate, ok := value.(*ViewUpdate)
+		if !ok {
+			return propagate, fmt.Errorf(
+				"failed to cast into view update, expected: *ViewUpdate, got: %T",
+				value)
+		}
+		nbr.handleViewUpdate(sender, *viewUpdate)
+
 	}
 
 	return propagate, nil
 }
 
 func (nbr *NetworkBridgeReceiver) handleValidationMessage(
 	sender peer.ID, msg network.NotificationsMessage) (bool, error) {
-	// we don't propagate collation messages, so it will always be false
+
 	propagate := false
 
 	if msg.Type() != network.ValidationMsgType {
 		return propagate, fmt.Errorf("%w, expected: %d, found:%d", ErrUnexpectedMessageOnValidationProtocol,
 			network.ValidationMsgType, msg.Type())
 	}
 
-	validationProtocol, ok := msg.(*validationprotocol.ValidationProtocol)
+	wireMessage, ok := msg.(*WireMessage)
 	if !ok {
-		return propagate, fmt.Errorf(
-			"failed to cast into collator protocol message, expected: *CollationProtocol, got: %T",
-			msg)
+		return propagate, fmt.Errorf("failed to cast into wire message, expected: *WireMessage, got: %T", msg)
 	}
 
-	nbr.SubsystemsToOverseer <- events.PeerMessage[validationprotocol.ValidationProtocol]{
-		PeerID:  sender,
-		Message: *validationProtocol,
+	index, value, err := wireMessage.IndexValue()
+	if err != nil {
+		return propagate, fmt.Errorf("getting index value: %w", err)
+	}
+
+	switch index {
+	case 1:
+		validationProtocol, ok := msg.(*validationprotocol.ValidationProtocol)
+		if !ok {
+			return propagate, fmt.Errorf(
+				"failed to cast into validation protocol message, expected: *ValidationProtocol, got: %T",
+				value)
+		}
+		nbr.SubsystemsToOverseer <- events.PeerMessage[validationprotocol.ValidationProtocol]{
+			PeerID:  sender,
+			Message: *validationProtocol,
+		}
+	case 2:
+		viewUpdate, ok := value.(*ViewUpdate)
+		if !ok {
+			return propagate, fmt.Errorf(
+				"failed to cast into view update, expected: *ViewUpdate, got: %T",
+				value)
+		}
+		nbr.handleViewUpdate(sender, *viewUpdate)
 	}
 
 	return propagate, nil
 }
 
+func (nbr *NetworkBridgeReceiver) handleViewUpdate(peer peer.ID, view ViewUpdate) error {
+
+	peerData, ok := nbr.peerData[peer]
+	if !ok {
+		return errors.New("peer not found")
+	}
+	if len(view.Heads) > newMaxHeads || view.FinalizedNumber < peerData.view.FinalizedNumber {
+		nbr.net.ReportPeer(peerset.ReputationChange{
+			Value:  peerset.CostMajor,
+			Reason: "malformed view",
+		}, peer)
+	} else if len(view.Heads) == 0 {
+		nbr.net.ReportPeer(peerset.ReputationChange{
+			Value:  peerset.CostMinor,
+			Reason: "peer sent us empty view",
+		}, peer)
+	} else if events.View(view).Equals(events.View(peerData.view)) {
```

Comment:
It's a bit confusing to have two `View` types here. I think on the call you mentioned that one is an alias of the other and it might be possible to use only one of them everywhere.

Reply:
Instead of adding an `Equal` method, I could have just used `checkHeadsEqual`. Did that in latest commit. That makes things lot more readable and understandable.